### PR TITLE
updated the vet clinic database

### PR DESCRIPTION
**Pull request by Gleennkar**
## Query and Updated Animals Table;
- Added a new column species and updated it to unspecified.
- Insert more data into the table.
- Updated the various directories.
- Deleted, Updated, Created the Savepoint and ROLLBACK data on the Table.
- Commited Changes carried out on the table
- Find below the various screenshoots.

Day 2

Day 2

Day 2

Day 2

Day 2

Day 2

Day 2 Changes

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -7,4 +7,39 @@ SELECT date_of_birth FROM animals WHERE name ='piachu' OR name ='Agumon';
...
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN '10.4' AND '17.3';
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN '10.4' AND '17.3';
+
+BEGIN;
+  update animals
+  set species = 'unspecified';
+  ROLLBACK;
```
Comment (Davidosky007) [inline queries.sql:15]: * Kindly verify that the change was made after `species = unspecified` and verify that the species columns went back to the state before the transaction.
 image

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,39 @@ SELECT date_of_birth FROM animals WHERE name ='piachu' OR name ='Agumon';
...
+
+  BEGIN;
+  update animals
+  set species = 'digimon' where name like '%mon';
+  update animals
+  set species = 'pokemon' where name not like '%mon';
```
Comment (Davidosky007) [inline queries.sql:21]: * Kindly verify that the change was made  here as requested in the[ project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Verify%20that%20change%20was%20made%20and%20persists%20after%20commit.)

File: queries.sql [hunk 3]
```diff
@@ -7,4 +7,39 @@ SELECT date_of_birth FROM animals WHERE name ='piachu' OR name ='Agumon';
...
+  COMMIT;
```
Comment (Davidosky007) [inline queries.sql:22]: * Kindly `verify that change persists after commit`

File: queries.sql [hunk 4]
```diff
@@ -7,4 +7,39 @@ SELECT date_of_birth FROM animals WHERE name ='piachu' OR name ='Agumon';
...
+  SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+  COMMIT;
+  
+
+SELECT count(*) FROM animals;
+SELECT count(*) FROM animals WHERE escape_attempts > 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) from animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-30' GROUP BY species;
```
Comment (Davidosky007) [inline queries.sql:45]: * Kindly adjust this query `WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-30'` as any animal born on the 31st of dec 2000 wont be counted so the correct query should be `WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'`

File: schema.sql [hunk 5]
```diff
@@ -8,5 +8,6 @@ CREATE TABLE animals (
   date_of_birth DATE,
   escape_attempts INTEGER,
   neutered BOOLEAN,
-  weight_kg DECIMAL
+  weight_kg DECIMAL,
+  species VARCHAR(50)
```
Comment (Davidosky007) [inline schema.sql]: * It is not best practice to manually add columns to your table Kindly do that using `ALTER TABLE table_name
ADD column_name datatype;` [Read more here ](https://www.w3schools.com/sql/sql_alter.asp)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Gleennkar 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Proper use of Gitflow 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gleennkar** [comment]
* Hello @Davidosky007,*
*  Hope my messages you well.*
*If you check the screenshots attached you will see the comments you
highlighted has been addressed. Thanks*

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello  @Gleennkar      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Gitflow is followed
- [x] Descriptive P.R
- [x] Descriptive README file




 

- [x] No Linter Errors







Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**